### PR TITLE
chore: fix test for `slice`

### PR DESCRIPTION
**Pull request by tserg**
### What I did

Fix failing fuzzing [test](https://github.com/vyperlang/vyper/actions/runs/6318947122/job/17158979469) for `slice`.

### How I did it

Make the test contract for each bytesdata location more specific.

### How to verify it

See tests.

### Commit message

```
chore: fix test for `slice``
```

### Description for the changelog

Fix test for `slice`.

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: tests/parser/functions/test_slice.py [hunk 1]
```diff
@@ -146,15 +152,19 @@ def _get_contract():
...
+            location == "literal"
+            and len(bytesdata) > length_bound
+            and ((literal_length and length > length_bound) or (not literal_length))
+        )
         or (literal_start and start > data_length)
         or (literal_length and length < 1)
     ):
         assert_compile_failed(lambda: _get_contract(), (ArgumentException, TypeMismatch))
-    elif len(bytesdata) > data_length:
+    elif location in "code" and len(bytesdata) > data_length:
```
Comment (charles-cooper) [inline tests/parser/functions/test_slice.py:164]: this test looks a bit sus. does this ever evaluate to true?

Comment (tserg) [reply tests/parser/functions/test_slice.py:164]: Yes, e.g. `location="code", bytesdata=b"\x00\x00", ..., literal_start=False, ..., literal_length=False, length_bound=1`

```
IMMUTABLE_BYTES: immutable(Bytes[1])
@external
def __init__(foo: Bytes[1]):
    IMMUTABLE_BYTES = foo

@external
def do_slice(inp: Bytes[1], start: uint256, length: uint256) -> Bytes[1]:
    return slice(IMMUTABLE_BYTES, start, length)
```

Comment (tserg) [reply tests/parser/functions/test_slice.py:164]: oh oops I see what the issue is now :see_no_evil:

File: tests/parser/functions/test_slice.py [hunk 2]
```diff
@@ -146,15 +152,19 @@ def _get_contract():
     if (
         (start + length > data_length and literal_start and literal_length)
         or (literal_length and length > data_length)
-        or (location == "literal" and len(bytesdata) > length_bound)
+        or (
+            location == "literal"
+            and len(bytesdata) > length_bound
+            and ((literal_length and length > length_bound) or (not literal_length))
```
Comment (charles-cooper) [inline tests/parser/functions/test_slice.py:158]: can this just be
```suggestion
            and (literal_length and length > length_bound)
```

Comment (tserg) [reply tests/parser/functions/test_slice.py:158]: No, this is a falsifying example (which I originally thought was a bug):
```
@external
def do_slice(inp: Bytes[1], start: uint256, length: uint256) -> Bytes[1]:
    return slice(b'\x00\x00', 0, length)
```

File: tests/parser/functions/test_slice.py [hunk 3]
```diff
@@ -146,15 +152,19 @@ def _get_contract():
     if (
         (start + length > data_length and literal_start and literal_length)
         or (literal_length and length > data_length)
-        or (location == "literal" and len(bytesdata) > length_bound)
+        or (
```
Comment (charles-cooper) [inline tests/parser/functions/test_slice.py:155]: from discussion offline: this large condition to be rewritten/simplified

**Review by charles-cooper — APPROVED** [review]
lgtm. will merge in a bit

**Pull request merged**